test/python/tests/csi/test_bdd_csi.py

```diff
@@ -86,63 +82,27 @@ def get_volume_capability(volume, read_only):
 
 
 @pytest.fixture(scope="module")
-def start_csi_plugin():
-    def monitor(proc, result):
-        stdout, stderr = proc.communicate()
-        result["stdout"] = stdout.decode()
-        result["stderr"] = stderr.decode()
-        result["status"] = proc.returncode
-
-    proc = subprocess.Popen(
-        args=[
-            "sudo",
-            os.environ["SRCDIR"] + "/target/debug/mayastor-csi",
-            "--csi-socket=/tmp/csi.sock",
-            "--grpc-endpoint=0.0.0.0",
-            "--node-name=msn-test",
-            "-v",
-        ],
-        stdout=subprocess.PIPE,
-        stderr=subprocess.PIPE,
-    )
-
-    result = {}
-    handler = threading.Thread(target=monitor, args=[proc, result])
-    handler.start()
-    time.sleep(1)
-    yield
-    subprocess.run(["sudo", "pkill", "mayastor-csi"], check=True)
-    handler.join()
-    print("[CSI] exit status: %d" % (result["status"]))
-    print(result["stdout"])
-    print(result["stderr"])
-
-
-@pytest.fixture(scope="module")
-def handles_mod(docker_project, module_scoped_container_getter):
-    assert "ms0" in docker_project.service_names
-    handles = {}
-    handles["ms0"] = MayastorHandle(
-        module_scoped_container_getter.get("ms0").get(
-            "NetworkSettings.Networks.mayastor_net.IPAddress"
-        )
+def mayastor_instance(container_mod):
+    container = container_mod.get("ms0")
+    yield MayastorHandle(
+        container.get("NetworkSettings.Networks.mayastor_net.IPAddress")
     )
-    yield handles
 
 
 @pytest.fixture(scope="module")
-def mayastor_instance(handles_mod):
-    yield handles_mod["ms0"]
+def csi_container(container_mod):
+    container = container_mod.get("ms1")
+    yield container.get("NetworkSettings.Networks.mayastor_net.IPAddress")
```

Comment:
Does it yield container itself or the container's IP address ? In case of IP address, I think we can rename the fixture to 'csi_container_address' or so, to make it more consistent, IMHO.

Reply:
It is actually used to enforce dependencies and so the return value is not really useful. Will change so that it does not return a value.